### PR TITLE
[ROCm][Windows] Fix Windows DLL linkage for batch norm (`-Winconsistent-dllimport`)

### PR DESCRIPTION
**Pull request by tvukovic-amd**
Fixes `warning: 'at::native::_get_cudnn_batch_norm_reserve_space_size' redeclared without 'dllimport' attribute: 'dllexport' attribute added [-Winconsistent-dllimport]`.

Fix: `aten/src/ATen/native/cudnn/BatchNorm.h`: change the forward declaration from `TORCH_API` to `TORCH_CUDA_CU_API` for `_get_cudnn_batch_norm_reserve_space_size`.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang

### CODE REVIEW

**Review by jeffdaily — APPROVED** [review]
Do we even use this file for ROCm?  Approved, pending CI.
